src/transformers/models/swiftformer/modeling_swiftformer.py

```diff
@@ -0,0 +1,638 @@
+
+    def __init__(self, config: SwiftFormerConfig, index, norm_layer=nn.BatchNorm2d):
+        super().__init__()
+
+        patch_size = config.down_patch_size
+        stride = config.down_stride
+        padding = config.down_pad
+        embed_dims = config.embed_dims
+
+        in_chans = embed_dims[index]
+        embed_dim = embed_dims[index + 1]
+
+        patch_size = patch_size if isinstance(patch_size, collections.abc.Iterable) else (patch_size, patch_size)
+        stride = stride if isinstance(stride, collections.abc.Iterable) else (stride, stride)
+        padding = padding if isinstance(padding, collections.abc.Iterable) else (padding, padding)
+
+        self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_size, stride=stride, padding=padding)
+        self.norm = norm_layer(embed_dim, eps=config.batch_norm_eps) if norm_layer else nn.Identity()
+
+    def forward(self, x):
+        x = self.proj(x)
+        x = self.norm(x)
+        return x
+
+
+class SwiftFormerConvEncoder(nn.Module):
+    """
+    Implementation of SwiftFormerConvEncoder with 3*3 and 1*1 convolutions. Input: tensor with shape [B, C, H, W]
+    Output: tensor with shape [B, C, H, W]
+    """
+
+    def __init__(
+        self, config: SwiftFormerConfig, dim, hidden_dim=64, kernel_size=3, drop_path=0.0, use_layer_scale=True
+    ):
+        super().__init__()
+        self.depth_wise_conv = nn.Conv2d(dim, dim, kernel_size=kernel_size, padding=kernel_size // 2, groups=dim)
+        self.norm = nn.BatchNorm2d(dim, eps=config.batch_norm_eps)
+        self.point_wise_conv1 = nn.Conv2d(dim, hidden_dim, kernel_size=1)
+        self.act = nn.GELU()
+        self.point_wise_conv2 = nn.Conv2d(hidden_dim, dim, kernel_size=1)
+        self.drop_path = SwiftFormerDropPath(drop_path) if drop_path > 0.0 else nn.Identity()
+        self.use_layer_scale = use_layer_scale
+        if use_layer_scale:
+            self.layer_scale = nn.Parameter(torch.ones(dim).unsqueeze(-1).unsqueeze(-1), requires_grad=True)
+
+    def forward(self, x):
+        input = x
+        x = self.depth_wise_conv(x)
+        x = self.norm(x)
+        x = self.point_wise_conv1(x)
+        x = self.act(x)
+        x = self.point_wise_conv2(x)
+        if self.use_layer_scale:
+            x = input + self.drop_path(self.layer_scale * x)
+        else:
+            x = input + self.drop_path(x)
+        return x
+
+
+class SwiftFormerMlp(nn.Module):
+    """
+    Implementation of MLP layer with 1*1 convolutions. Input: tensor with shape [batch_size, channels, height, width]
+    Output: tensor with shape [batch_size, channels, height, width]
+    """
+
+    def __init__(
+        self,
+        config: SwiftFormerConfig,
+        in_features,
+        hidden_features=None,
+        out_features=None,
+        act_layer=nn.GELU,
+        drop=0.0,
+    ):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
```

Comment:
We prefer to explicitly check whether a value is None

```suggestion
        out_features = out_features if out_features is not None else in_features
        hidden_features = hidden_features if hidden_features is not None else in_features
```